spec/jobs/petition_pdf_generation_worker_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe PetitionPdfGenerationWorker do
+  let(:petition_pdf_service) { spy }
+  let(:worker) { described_class.new petition_pdf_service: petition_pdf_service }
+
+  describe "#perform" do
+    let!(:cycle) { create_cycle_with_phase phases: [{ plugin_type: :petition }] }
+    let!(:phase) { cycle.phases.first }
+    let!(:detail) { create :petition_plugin_detail, plugin_relation: phase.plugin_relation }
+    let!(:version) { create :petition_plugin_detail_version, petition_plugin_detail: detail, published: false }
+
+    let(:document_url) { "https://teste.s3-us-west-2.amazonaws.com/seguranca-publica-peticao-1-1.pdf" }
+
+    subject { worker.perform nil, "{ \"id\": #{version.id} }" }
+
+    before { allow(petition_pdf_service).to receive(:generate).and_return document_url }
```

Comment:
For example if somebody removed the `generate` method from this service, this test would not fail the way it is now. So `instance_spy` to the rescue!